test/e2e/storage/utils/ebs.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+const (
+	volumeAttachmentStatusPollDelay = 2 * time.Second
+	volumeAttachmentStatusFactor    = 2
+	volumeAttachmentStatusSteps     = 3
```

Comment:
It might be too long to wait in e2e tbh. A test that takes too long to run is a useless test. for example - if any of e2e really hits 20 minute wait, then that test will simply timeout (pod startup timeout is 5mins) and fail. I think - we should be fine with it failing in those cases.

Reply:
Sure, 20 minutes is too long, but in contrast 2*2^3 (=16) seconds might be too short. What about 2 minutes (6 steps, 128 seconds, if I am counting right?)

Reply:
okay increased this to 6 steps, which would be total of 126s.